cf-custom-resources/lib/copy-assets.js

```diff
@@ -11,11 +11,27 @@ const aws = require("aws-sdk");
 exports.handler = async function (event, context, callback) {
 	const s3 = new aws.S3();
 
-	const params = {
+	try {
+		await s3.headObject({
+			Bucket: event.destBucket,
+			Key: event.mapping.destPath
+		}).promise();
+
+		// if headObject is successful, the file already exists
+		// in destBucket. no need to copy, so we're done!
```

Comment:
just checking: what about in an 'update' situation? the file exists but a newer version needs to be copied....

Reply:
good catch😅 let me see what we can do about that!

Reply:
ok i think i'm actually going to revert these changes.

we _could_ add a `headObject` call to the source object, and only do the upload if the destination file does not exist (what we have now) _**or**_ if the [files etag's](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Object.html) are mismatched. however, since [copying data between two s3 buckets the same region is free](https://aws.amazon.com/s3/pricing/), i think it's simpler & faster to just remove that logic and let s3 do the copy even if it's unnecessary.